lib/iris/tests/unit/fileformats/netcdf/test_parse_cell_methods.py

```diff
@@ -85,6 +99,53 @@ def test_comment(self):
             res = parse_cell_methods(cell_method_str)
             self.assertEqual(res, expected)
 
+    def test_comment_brackets(self):
+        cell_method_strings = [
+            "time: minimum within days (comment: 18h(day-1)-18h)",
+            "time : minimum within days (comment: 18h(day-1)-18h)",
+        ]
+        expected = (
+            CellMethod(
+                method="minimum within days",
+                coords="time",
+                intervals=None,
+                comments="18h(day-1)-18h",
+            ),
+        )
+        for cell_method_str in cell_method_strings:
+            res = parse_cell_methods(cell_method_str)
+            self.assertEqual(res, expected)
+
+    def test_comment_bracket_mismatch_warning(self):
+        cell_method_strings = [
+            "time: minimum within days (comment: 18h day-1)-18h)",
+            "time : minimum within days (comment: 18h day-1)-18h)",
+        ]
+        for cell_method_str in cell_method_strings:
+            with mock.patch("warnings.warn") as warn:
```

Comment:
Would `self.assertWarnsRegexp` do the job here?

https://github.com/SciTools/iris/blob/91b01be11ebd439668c3f140f66350b9e65db59e/lib/iris/tests/__init__.py#L601-L609

Reply:
I'd gone for copying the code below what I'd inserted (below) but don't know if one of these is more up-to-date or preferred. I think either works?

https://github.com/SciTools/iris/blob/a7869293fb8a94765a2508ff126dfd1fd3f42679/lib/iris/tests/unit/fileformats/netcdf/test_parse_cell_methods.py#L187-L193

Reply:
Looking at the git blame I _think_ the `assertWarnsRegexp` is newer.  I generally prefer to have a single function/method that is called in multiple places, rather than copy/paste.  That way, if I think of a better way to do it later, I only need to change it in one place to get the improvement all over.

I would defer to alternative advice from anyone in your team though!

Reply:
Turns out assertWarns takes a msg argument, so I've used that in both places. Thanks for the prompt to dig into this!

Reply:
Looks like [assertWarns and assertWarnsRegex were introduced into unittest at python3.2](https://docs.python.org/3/library/unittest.html#unittest.TestCase.assertWarns).  Our own `assertWarnsRegex` was presumably written when we still supported python2.7, and is now redundant 🤔

Reply:
Ah that explains why assertWarnsRegex and assertWarnsRegexp were both showing up as options in VSCode